hubblestack/extmods/grains/fqdn.py

```diff
@@ -19,5 +21,60 @@ def fqdn():
     local_fqdn = None
     if not salt.utils.platform.is_windows():
         local_fqdn = __salt__['cmd.run']('hostname --fqdn')
-    grains['local_fqdn'] = local_fqdn if local_fqdn else socket.getfqdn()
+    if not local_fqdn or 'hostname: ' in local_fqdn:
+        grains['local_fqdn'] = ''
+    else:
+        grains['local_fqdn'] = local_fqdn
     return grains
+
+
+def dest_ip():
+    '''
+    Generate a best-guess at the IP on the interface that is the default
+    gateway for the host. This is because the current methods can result in
+    various IPs due to round robin DNS.
+    '''
+    grains = {}
+    interfaces = salt.grains.core.ip4_interfaces()['ip4_interfaces']
+    try:
+        ret = __salt__['cmd.run_all']('ip route show to 0/0')
+        if ret['retcode'] == 0:
+            interface = None
+            try:
+                interface = ret['stdout'].split(' ')[4]
```

Comment:
Just for the remainder and documentation for the next iteration, in the output of ```ip route``` device is always preceded by the ```dev``` keyword.